### PR TITLE
CB-15150 Node status should be unhealthy when it is running in cloudp…

### PR DESCRIPTION
**Pull request by cegganesh84**
…rovider, but in maintenance mode in CM

1. In failure scenarios of stop/start scaling it may happen that the CM commissioning was not successful.
2. In those scenarios customer need to know that the host status is bad, so that they can take corrective actions.
3. That said, should this maintenance mode checking need to be under stop/start entitlement, and why was it ignored in other cases?

See detailed description in the commit message.

### CODE REVIEW

**Review by horadla23 — APPROVED** [review]
LGTM

**Review by aakashshankar — APPROVED** [review]

**Review by sidseth — APPROVED** [review]
Applicable 7.2.12 onwards only - per the extended host health status. Suspect at least maintenance mode information is available earlier itself. Can be chased in a follow-up.

**Comment by sidseth** [comment]
Also holding off on the merge until CB-14929 is merged to master, since that is under active review.

**Pull request merged**
